core/src/main/java/io/undertow/UndertowOptions.java

```diff
@@ -77,6 +77,18 @@
      */
     public static final Option<Integer> NO_REQUEST_TIMEOUT = Option.simple(UndertowOptions.class, "NO_REQUEST_TIMEOUT", Integer.class);
 
+    /**
+     * Default value( miliseconds ) for async context timeout.
+     * @see {@link javax.servlet.AsyncContext#setTimeout}
+     */
+    public static final Option<Long> ASYNC_CONTEXT_TIMEOUT = Option.simple(UndertowOptions.class, "ASYNC_CONTEXT_TIMEOUT", Long.class);
+
+    /**
+     * Default value of {@link #ASYNC_CONTEXT_TIMEOUT}
+     * @see {@link javax.servlet.AsyncContext#setTimeout}
+     */
+    public static final long DEFAULT_ASYNC_CONTEXT_TIMEOUT = 30000;
```

Comment:
I think `ASYNC_CONTEXT_TIMEOUT` and `DEFAULT_ASYNC_CONTEXT_TIMEOUT` are not needed as UndertowOptions. If I understand correctly, `UndertowOptions` is basically used for specifying the listener-level options.
I think just adding an API to `DeploymentInfo` is enough for the deployment-level option because WildFly (JBoss EAP) can invoke the `DeploymentInfo` API from `UndertowDeploymentInfoService` to configure `<servlet-container>` setting as the default deployment-level options.